sw/device/silicon_creator/mask_rom/mask_rom.c

```diff
@@ -50,168 +54,115 @@ static inline rom_error_t mask_rom_irq_error(void) {
   return kErrorInterrupt + mcause;
 }
 
-void mask_rom_interrupt_handler(void) {
-  shutdown_finalize(mask_rom_irq_error());
-}
-
-// We only need a single handler for all mask ROM interrupts, but we want to
-// keep distinct symbols to make writing tests easier.  In the mask ROM,
-// alias all interrupt handler symbols to the single handler.
-void mask_rom_exception_handler(void)
-    __attribute__((alias("mask_rom_interrupt_handler")));
-
-void mask_rom_nmi_handler(void)
-    __attribute__((alias("mask_rom_interrupt_handler")));
-
-rom_error_t mask_rom_boot(void) {
+/**
+ * Performs once-per-boot initialization of mask ROM modules and peripherals.
+ */
+static void mask_rom_init(void) {
+  sec_mmio_init(shutdown_finalize);
   // Initialize the shutdown policy according to lifecycle state.
-  lifecycle_state_t lc_state = lifecycle_state_get();
+  lc_state = lifecycle_state_get();
   shutdown_init(lc_state);
-
-  // Initiaize shadow copy of the ePMP register configuration.
-  epmp_state_t epmp;
+  // Initiaize in-memory copy of the ePMP register configuration.
   mask_rom_epmp_state_init(&epmp);
-
   // Initialize pinmux configuration so we can use the UART.
   pinmux_init();
-
   // Configure UART0 as stdout.
   uart_init(kUartNCOValue);
   base_set_stdout((buffer_sink_t){
       .data = NULL,
       .sink = uart_sink,
   });
+  // TODO: Clean device state based on reset reason (chip-specific startup).
+  // TODO: Bootstrap.
 
   // FIXME: what (if anything) should we print at startup?
```

Comment:
Should the `printf`s be in `mask_rom_main`? The don't really feel like a good fit for `mask_rom_init` to me.

Reply:
They are related to initialization but I left them in `mask_rom_init()` since they are once-per-boot but I don't have a strong preference. Moved to `mask_rom_main()`.